### ISSUE TITLE
iPhoneXr and XsMax start render from (0, 24) on non-header page

### ISSUE THREAD

**Issue by FaiChou** [react-navigation/react-navigation#5205]
### Current Behavior

<img width="656" alt="xnip2018-11-05_16-26-09" src="https://user-images.githubusercontent.com/18500846/47986465-d04ced80-e117-11e8-9b6d-66128f35fbad.png">

<img width="642" alt="xnip2018-11-05_16-27-50" src="https://user-images.githubusercontent.com/18500846/47986514-f6728d80-e117-11e8-80c9-b76df9a57c10.png">

### Expected Behavior

<img width="562" alt="xnip2018-11-05_16-37-22" src="https://user-images.githubusercontent.com/18500846/47986941-1d7d8f00-e119-11e8-9a83-cc3430de5537.png">

### How to reproduce

The 0.7.0 and 0.8.0 version of `react-native-safe-area-view` were not suit for iPhoneXr and XsMax, so I change the source code of these two. Now the weird thing comes: the Xr and XsMax screen render from (0, 24) in react-navigation non-header page.

```
node_modules:
  react-native-safe-area-view@0.8.0
  react-navigation@2.2.0
  react-navigation-tabs@0.5.0
    node_modules:
      react-native-safe-area-view@0.7.0
```

What I found that contain `24` was in `react-native-safe-area-view/index.js` 

```javascript
      case 'bottom': {
        return isIPhoneX ? (isLandscape ? 24 : 34) : 0;
      }
```

So in conclusion:

1. react-navigation 2.2.0
2. iPhoneXr and iPhoneXsMax simulator
3. modified react-native-safe-area-view (both two versions)
4. no-navigationBar page

### Your Environment

Environment:
  OS: macOS 10.14.1
  Node: 9.5.0
  Yarn: 1.7.0
  npm: 5.7.1
  Watchman: 4.9.0
  Xcode: Xcode 10.1 Build version 10B61
  Android Studio: 3.1 AI-173.4720617

Packages: (wanted => installed)
  react: 16.3.1 => 16.3.1
  react-native: 0.55.4 => 0.55.4

**Comment by FaiChou** [comment]
### Why not upgrade react-navigation

My RN project is using `dva`, and 1.1.2 version of `react-navigation-redux-helpers` to manage state of react-navigation.

When I try to upgrade react-navigation, I have to upgrade redux-helpers to v2.x too.

The 1.x version helper's api `createNavigationPropConstructor` was deprecated on v2.x. The new api `reduxifyNavigator ` and `createReactNavigationReduxMiddleware` are not suitable for dva (as I know).

```javascript
  render() {
    const navigationPropConstructor = createNavigationPropConstructor('root');
    const navigation = navigationPropConstructor(
      (action) => {
        if (actions.indexOf(action.type) !== -1) {
          this.props.dispatch({ type: 'nav/apply', payload: action });
        } else if (action.type === 'backToRoot' || action.type === 'backTo') {
          this.props.dispatch({ type: `nav/${action.type}`, payload: action });
        }
      },
      this.props.nav,
    );
    return (
      <AppNavigator navigation={navigation} />
    );
  }
```

**Comment by brentvatne** [comment]
hi! sorry that you need redux helpers. we don't support redux helpers officially. if this is preventing you from updating then you should post an issue to that repository: https://github.com/react-navigation/react-navigation-redux-helpers

**Comment by FaiChou** [comment]
@brentvatne, you may not clear about my question: Xr and XsMax render wrong position of screen (from 0,24). And I cannot upgrade navigation, so how to fix it ?

When I look up the source code , I found https://github.com/react-navigation/react-navigation/blob/2.2.0/src/views/StackView/StackViewLayout.js#L29 where the trouble comes.

So change the code and fix it well by self.

**Comment by brentvatne** [comment]
@FaiChou - the solution is to update react-navigation. I understand that you cannot update it because react-navigation-redux-helpers doesn't support something you need anymore, so you need to go to that repository and ask them how you can accomplish what you're trying to do now that the library doesn't support it anymore :)

if isn't satisfactory for you then you could fork this repository and update that line of code :) but i'd suggest updating react-navigation and then trying to figure out how to do what you need to with the new version of react-navigation-redux-helpers

**Comment by FaiChou** [comment]
@brentvatne  thank you, I forked the project and using it by `git+https://github.com/.../.git#my-fixed-tag`, finally solved my problem. It is the worst way but time waits for nobody :)

**Comment by FaiChou** [comment]
Here's a bad solution for me: https://stackoverflow.com/a/53272312/10595695
